packages/http/src/validator/validators/body.ts

```diff
@@ -54,6 +54,15 @@ export function splitUriParams(target: string) {
   }, {});
 }
 
+export function parseMultipartFormDataParams(target: string) {
+  const pairs = target.split(/--.*\s*.*="/).filter(element => element);
```

Comment:
I think this code assumes that the boundary between multipart entities is `--` followed by _any_ string.  However, the boundary between multipart entities is defined, in each individual request, by the request's `Content-Type` header's `Boundary` parameter.  This is because the client must choose a value that doesn't occur _within_ any multipart entity that it will transmit.  For example, consider the effect of the following example on your implementation.

```
curl -v -F a=b -F c='--="' <hostname>
```

Also consider any form value that includes `\r\n\r\n`.  

[RFC 2046 § 5.1.1](https://datatracker.ietf.org/doc/html/rfc2046#section-5.1.1) describes the boundary as...

> a line consisting entirely of two hyphen characters ("-", decimal value 45) followed by the boundary parameter value from the Content-Type header field, optional linear whitespace, and a terminating CRLF.

I think you'll need to either:
1. implement this as the standard describes, with unit tests (not `test-harness` integration tests) to validate that it works with a variety of input, OR
2. rely on a library which parses multipart data according to the standard.

Reply:
Hi Ed, thank you so much for your feedback. I discovered a library called parse-multiform-data which I'm installing as an npm package that seemed to be the most reliable method. I made changes according to this, such as parsing the boundary parameter from the Content-Type header field in order to pass it to the library for parsing the body. I left some comments on the recent changes that hopefully help clarify some of the decisions I made when implementing this. Thanks again and please let me know if you'd like any other changes!